adapters/webhdfs-adapter/src/main/java/com/connexta/replication/adapters/webhdfs/WebHdfsNodeAdapter.java

```diff
@@ -372,8 +375,7 @@ String getVersion4Uuid(String fileUrl, Date modificationTime) {
         results.add(file);
       } else {
         String fileUrl = getWebHdfsUrl().toString() + file.getPathSuffix();
-        Date modificationTime = file.getModificationTime();
-        String id = getVersion4Uuid(fileUrl, modificationTime);
+        String id = getVersion4Uuid(fileUrl, file.getModificationTime());
```

Comment:
You could skip creating the `fileUrl` String and just pass the concatenated values directly.